src/main/kotlin/io/meshcloud/dockerosb/persistence/SynchronizedGitHandler.kt

```diff
@@ -209,36 +215,22 @@ class SynchronizedGitHandler(
   }
 
   private fun cleanUp() {
-      git.clean().setForce(true).setCleanDirectories(true).call()
-      git.reset().setMode(ResetCommand.ResetType.HARD).setRef("HEAD").call()
+    git.clean().setForce(true).setCleanDirectories(true).call()
```

Comment:
d: We should also apply this refactroing, especially the handling of the git lock, to our noop osb. Especially the git lock handling is important as the noop osb hangs because of that all the time.

Reply:
I think the noop-broker probably needs to do more and also remove the lock file after some time, as it seems to be a persistent error there. Here we only do a simple retry, that occurred from time to time, mostly when I observed the git repo in parallel as via command line as well.

I assume there error here is probably that the fs somehow too slow and lists the file as still there, when we have a lock already released in the code. So I do the retry and the lock file is gone then.

Reply:
ok, i guess then it's best to have a ticket for the noop sb fix, because i think we need it. Could you please create as you can provide the most detailed info for that ticket

Reply:
I think that this is actually already fixed with https://github.com/meshcloud/meshfed-release/pull/1109

Reply:
ah, nice, wasn't aware of that